pkg/api/plugin.go

```diff
@@ -18,6 +18,8 @@ type Plugin interface {
 
 	GenerateARM(cs *OpenShiftManagedCluster) ([]byte, error)
 
+	EnsureSyncPod(ctx context.Context, cs *OpenShiftManagedCluster) error
```

Comment:
This feels like it may grow into a lifecycle hook.  Perhaps something like `InitializeCluster` or `CreateAddons` gives us a little more leeway.

Reply:
done in #231